.tekton/build-service-pull-request.yaml

```diff
@@ -167,18 +167,83 @@ spec:
       workspaces:
       - name: basic-auth
         workspace: git-auth
+    - name: sealights-go-instrumentation
+      runAfter:
+        - clone-repository
+      taskRef:
+        params:
+        - name: name
+          value: sealights-go-oci-ta
+        - name: bundle
+          value: quay.io/konflux-ci/tekton-catalog/task-sealights-go-oci-ta:0.1@sha256:3ee87d28065205ee5f1e32ec012cac4b167af6525f7504a9657612e4387d0eae
+        - name: kind
+          value: task
+        resolver: bundles
+      params:
+        - name: SOURCE_ARTIFACT
+          value: $(tasks.clone-repository.results.SOURCE_ARTIFACT)
+        - name: go-version
+          value: "1.21.9"
+        - name: sealights-secret
+          value: "sealights-credentials"
+        - name: component
+          value: '{{ repo_name }}'
+        - name: branch
+          value: '{{ source_branch }}'
+        - name: revision
+          value: '{{ revision }}'
+        - name: repository-url
+          value: '{{ repo_url }}'
+        - name: pull-request-number
+          value: '{{ pull_request_number }}'
+        - name: target-branch
+          value: '{{ target_branch }}'
+        - name: oci-storage
+          value: $(params.output-image).sealights.git
+    - name: sealights-unit-tests
+      runAfter:
+        - sealights-go-instrumentation
+      taskSpec:
+        volumes:
+          - name: sealights-credentials
+            secret:
+              secretName: sealights-credentials
+          - name: workdir
+            emptyDir: {}
+        stepTemplate:
+          volumeMounts:
+            - mountPath: /var/workdir
+              name: workdir
+            - name: sealights-credentials
+              mountPath: /usr/local/sealights-credentials
+        steps:
+          - name: use-trusted-artifact
+            image: quay.io/redhat-appstudio/build-trusted-artifacts:latest@sha256:52f1391e6f1c472fd10bb838f64fae2ed3320c636f536014978a5ddbdfc6b3af
+            args:
+              - use
+              - $(tasks.sealights-go-instrumentation.results.SOURCE_ARTIFACT)=/var/workdir/source
+          - name: unit-tests
+            image: registry.access.redhat.com/ubi8/go-toolset:1.21.9
+            securityContext:
+              runAsUser: 0
+            workingDir: /var/workdir/source
+            script: |
+              #!/bin/bash
+              set -euo pipefail
+              export SEALIGHTS_TOKEN="$(cat /usr/local/sealights-credentials/token)"
+              make test
     - name: prefetch-dependencies
       params:
       - name: input
         value: $(params.prefetch-input)
       - name: SOURCE_ARTIFACT
-        value: $(tasks.clone-repository.results.SOURCE_ARTIFACT)
+        value: $(tasks.sealights-go-instrumentation.results.SOURCE_ARTIFACT)
```

Comment:
Is the `sealights-get-refs` task smart enough to find the "sealights-enabled" container even if the source artifact chain goes `clone -> sealights-instrumentation -> prefetch -> build`?

Reply:
I would say yes. For now is detecting the container the container test in the integration service. I see in the ui the tests are detecting sealights agents.
![Screenshot From 2025-02-03 10-31-03](https://github.com/user-attachments/assets/d812efd6-074d-4a6e-9e82-8a5b50b6b8c7)

https://github.com/konflux-ci/tekton-integration-catalog/blob/main/tasks/sealights/sealights-get-refs/0.1/sealights-get-refs.yaml#L66-L71